recipes/cpprestsdk/all/conanfile.py

```diff
@@ -0,0 +1,166 @@
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "exclude_websockets": False,
+        "exclude_compression": False
+    }
+    short_paths = True
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        if self.settings.compiler == 'Visual Studio':
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires.add("openssl/1.1.1e")
+        if not self.options.exclude_compression:
+            self.requires.add("zlib/1.2.11")
+        if not self.options.exclude_websockets:
+            self.requires.add("websocketpp/0.8.1")
+        self.requires.add("boost/1.72.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        if self.settings.os == "iOS":
+            with open('toolchain.cmake', 'w') as toolchain_cmake:
+                if self.settings.arch == "armv8":
+                    arch = "arm64"
+                    sdk = "iphoneos"
+                elif self.settings.arch == "x86_64":
+                    arch = "x86_64"
+                    sdk = "iphonesimulator"
+                sysroot = tools.XCRun(self.settings).sdk_path
+                toolchain_cmake.write('set(CMAKE_C_COMPILER /usr/bin/clang CACHE STRING "" FORCE)\n')
+                toolchain_cmake.write('set(CMAKE_CXX_COMPILER /usr/bin/clang++ CACHE STRING "" FORCE)\n')
+                toolchain_cmake.write('set(CMAKE_C_COMPILER_WORKS YES)\n')
+                toolchain_cmake.write('set(CMAKE_CXX_COMPILER_WORKS YES)\n')
+                toolchain_cmake.write('set(CMAKE_XCODE_EFFECTIVE_PLATFORMS "-%s" CACHE STRING "" FORCE)\n' % sdk)
+                toolchain_cmake.write('set(CMAKE_OSX_ARCHITECTURES "%s" CACHE STRING "" FORCE)\n' % arch)
+                toolchain_cmake.write('set(CMAKE_OSX_SYSROOT "%s" CACHE STRING "" FORCE)\n' % sysroot)
+            os.environ['CONAN_CMAKE_TOOLCHAIN_FILE'] = os.path.join(os.getcwd(), 'toolchain.cmake')
+
+        self._cmake = CMake(self, set_cmake_flags=True)
+        self._cmake.definitions["BUILD_TESTS"] = False
+        self._cmake.definitions["BUILD_SAMPLES"] = False
+        self._cmake.definitions["WERROR"] = False
+        self._cmake.definitions["CPPREST_EXCLUDE_WEBSOCKETS"] = self.options.exclude_websockets
+        self._cmake.definitions["CPPREST_EXCLUDE_COMPRESSION"] = self.options.exclude_compression
+        if self.settings.os == "iOS":
+            self._cmake.definitions["IOS"] = True
+        elif self.settings.os == "Android":
+            self._cmake.definitions["ANDROID"] = True
+            self._cmake.definitions["CONAN_LIBCXX"] = ''
```

Comment:
What is the reason for this?
Maybe a dependency sets wrong flags?

Reply:
Copy paste from bincrafters, I have no clue =)

Reply:
@SSE4 
*only 2 years ago*
https://github.com/bincrafters/conan-cpprestsdk/commit/c648cea071b982a78f7e3901b499f8d3229ec929 😈😄 
Do you have any clues why this is/was needed?

I would comment the line out (+add comment) and see what happens.

Reply:
From the wiki, Android is not even listed. Is supporting android require for CCI?

Reply:
It is not tested by CI, but patches are accepted as mobile devices are popular these days. (What is the official stance? @uilianries)

When you think about it, the `IOS` and `ANDROID` lines can be removed as well.
See https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html#variables-that-describe-the-system

Reply:
When I search for the string "android" in their github cpprestsdk repo, I can find some hits.
So it is supported by the project (to some level at least)

Reply:
Android support came from another contributor. Need to investigate if that recipe really requires those definitions

Reply:
I have opted for removing Android. If in the future it's needed we can always revisit the topic.